marklogic-client-api/src/test/java/com/marklogic/client/test/dataservices/CallbackGeneratorTest.java

```diff
@@ -76,20 +96,36 @@ public static void setup() {
        
     }
 
-    @Test(expected = Test.None.class)
+    @Test
     public void forArgsGeneratorTest() {
-        
+        List<Double> outputValues = new ArrayList<Double>();
+        class Output {
+            List<Double> inputValues = new ArrayList<Double>();
+            int i;
+            Output() {
+                inputValues = Stream
+                        .iterate(10.0, last -> last - Double.valueOf(1))
+                        .limit(10)
+                        .collect(Collectors.toList());
+                Collections.sort(inputValues);
+                i = 0;
+            }
+        }
+        final Output output = new Output();
         batcher = caller
                 .batcher()
-                .forArgsGenerator(result -> (result == null) ? caller.args().param("param1", 1.1) : result.getArgs())
+                .forArgsGenerator(result -> (result == null || output.i<output.inputValues.size()) ? caller.args().param("param1", output.inputValues.get(output.i)) : null)
                 .onCallSuccess(event -> {
-                   assertEquals(event.getItem(), Double.valueOf(1.1));
+                    output.i+= 1;
```

Comment:
Do we need to count the number of successful calls, given that we are using a list instead of a set?  On the grounds that a list allows for duplicates so the size of the list should be the same as output.i?